compiler/quilt/tools/util.py

```diff
@@ -377,3 +378,25 @@ def find_in_package(corenode, path):
         traversed = '/'.join(path[:count])
         raise TypeError("Not a GroupNode: Node at {!r}".format(traversed))
 
+def save(user: str,
+         package: str,
+         data_node: str,
+         data: Union[np.ndarray, pd.DataFrame],
+         params: dict):
+    """Build and push data to Quilt registry at user/package/data_node,
+    associating params as metadata for the data node.
+    :param user: user name
+    :param package: package name
+    :param data_node: data node name
+    :param data: data to save
+    :param params: metadata
+    """
+    quilt.build(f'{user}/{package}/{data_node}', data)
```

Comment:
These are not compatible with Python 2.7; and I think that's why CI is blowing up. I _think_ `"{}/{}/{}".format(...)` will work

Reply:
Typing also isn't supported in Python2.